appinventor/components/src/com/google/appinventor/components/runtime/Ev3Motors.java

```diff
@@ -0,0 +1,664 @@
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+import com.google.appinventor.components.runtime.util.Ev3Constants;
+import com.google.appinventor.components.runtime.util.Ev3BinaryParser;
+import com.google.appinventor.components.runtime.util.ErrorMessages;
+import android.os.Handler;
+
+/**
+ * A component that provides both high- and low-level interfaces to
+ * control the motors on LEGO MINDSTORMS EV3.
+ *
+ * @author jerry73204@gmail.com (jerry73204)
+ * @author spaded06543@gmail.com (Alvin Chang)
+ */
+@DesignerComponent(version = YaVersion.EV3_MOTORS_COMPONENT_VERSION,
+                   description = "A component that provides both high- and low-level interfaces to a LEGO MINDSTORMS EV3 " +
+                                 "robot, with functions that can control the motors.",
+                   category = ComponentCategory.LEGOMINDSTORMS,
+                   nonVisible = true,
+                   iconName = "images/legoMindstormsEv3.png")
+@SimpleObject
+public class Ev3Motors extends LegoMindstormsEv3Base {
+  private static final int DELAY_MILLISECONDS = 50;
+  private static final String DEFAULT_MOTOR_PORT = "ABC";
+  private static final double DEFAULT_WHEEL_DIAMETER = 4.32;
+
+  private int motorPortBitField = 1;
+  private double wheelDiameter = DEFAULT_WHEEL_DIAMETER;
+  private boolean regulationEnabled = true;
+  private boolean stopBeforeDisconnect = true;
+  private boolean tachoCountChangedEventEnabled = false;
+  private final Runnable sensorValueChecker;
+  private Handler eventHandler;
+  private int previousValue = 0;
+  private boolean ifReset = false;
+
+  /**
+   * Creates a new Ev3Motors component.
+   */
+  public Ev3Motors(ComponentContainer container) {
+    super(container, "Ev3Motors");
+    eventHandler = new Handler();
+    sensorValueChecker = new Runnable() {
+      public void run() {
+        String functionName = "";
+
+        if (bluetooth != null && bluetooth.IsConnected()) {
+          int sensorValue = getOutputCount(functionName, 0, motorPortBitField);
+
+          if (!ifReset) {
+            if (sensorValue != previousValue && tachoCountChangedEventEnabled) {
+              TachoCountChanged(sensorValue);
+            }
+          } else {
+            ifReset = false;
+          }
+          previousValue = sensorValue;
+        }
+
+        eventHandler.postDelayed(this, DELAY_MILLISECONDS);
+      }
+    };
+
+    eventHandler.post(sensorValueChecker);
+
+    MotorPort(DEFAULT_MOTOR_PORT);
+    StopBeforeDisconnect(true);
+    EnableSpeedRegulation(true);
+    WheelDiameter(DEFAULT_WHEEL_DIAMETER);
+    TachoCountChangedEventEnabled(false);
+  }
+
+  /**
+   * Returns the motor port.
+   */
+  @SimpleProperty(description = "The motor ports that the motors are connected to. The ports are specified by a sequence of port letters.",
+                  category = PropertyCategory.BEHAVIOR)
```

Comment:
I think you should add "userVisible = false" so the app can't change the motor ports at runtime. It should only be set from the designer.